controllers/barbicanapi_controller.go

```diff
@@ -197,15 +198,19 @@ func (r *BarbicanAPIReconciler) getSecret(
 	secretName string,
 	envVars *map[string]env.Setter,
 ) (ctrl.Result, error) {
-	hash, result, err := secret.VerifySecret(ctx, types.NamespacedName{Name: instance.Spec.Secret, Namespace: instance.Namespace}, []string{"AdminPassword"}, h.GetClient(), time.Second*10)
+	expectedField := "transport_url"
+	if strings.HasSuffix(secretName, "osp-secret") {
+		expectedField = "AdminPassword"
```

Comment:
Although you are just reusing this existing string, hard-coding  `"AdminPassword"` here is likely a bug.  Every secret used, and they keys needed within that secret to access the data we need should be defined in the spec, not hard-coded into the controller like it is currently.

This value should probably be replaced with `instance.Spec.PasswordSelectors.Service`, which is the password that we need for the Barbican service user.  The default value for that is `"BarbicanPassword"`, not `"AdminPassword"`.

https://github.com/openstack-k8s-operators/barbican-operator/blob/main/api/v1beta1/common_types.go#L107-L109

It seems things are currently working because install_yamls reuses the same password for both of these fields in `"osp-secret"`, but these should be different in a real world deployment.

https://github.com/openstack-k8s-operators/install_yamls/blob/main/scripts/gen-input-kustomize.sh#L65-L67

Given that we care about different keys depending on which secret is being retrieved, then I think it makes sense that the list of keys we care about should be passed into this method alongside the name of the relevant secret.